fairlearn/post_processing/roc_curve_based_post_processing.py

```diff
@@ -147,18 +154,27 @@ def _validate_estimator(self):
                 not callable(fit_function):
             raise ValueError(MISSING_FIT_PREDICT_ERROR_MESSAGE)
 
-    def _validate_input_data(self, X, protected_attribute, y=None):
-        if type(X) != type(protected_attribute) or (y is not None and type(X) != type(y)) or \
-                type(X) not in [list, np.ndarray]:
-            raise ValueError(INPUT_DATA_CONSISTENCY_ERROR_MESSAGE
-                             .format(type(X).__name__, type(y).__name__,
-                                     type(protected_attribute).__name__))
-
-        if len(X) == 0 or len(protected_attribute) == 0 or (y is not None and len(y) == 0):
+    def _validate_input_data(self, X, aux_data, y=None):
+        allowed_input_types = [list, np.ndarray, pd.DataFrame, pd.Series]
+        if type(X) not in allowed_input_types or \
+                type(aux_data) not in allowed_input_types or \
+                (y is not None and type(y) not in allowed_input_types):
```

Comment:
Same kind of thing as I'd mentioned before - this kind of line continuation can be really hard to follow in a conditional. We've tended to set a variable for each of the sections then have a one line conditional with those variables. Same benefit of the variable name helping document everything.

Reply:
To what extent would we get this 'for free' if we split the multiple-type support from interpret into a common library?

Reply:
But it reads like text!
if type of X not in allowed input types or type of aux_data not in allowed input types or type of y not in allowed input types. I could not possibly put that in three variables that make it clearer. 
```
X_not_in_allowed_input_types = type(X) not in allowed_input_types
aux_data_not_in_allowed_input_types = type(aux_data) not in allowed_input_types
y_not_in_allowed_input_types = (y is not None and type(y) not in allowed_input_types)

if X_not_in_allowed_input_types or aux_data_not_in_allowed_input_types or \
    y_not_in_allowed_input_types:
    ...
```
hardly seems better.

Reply:
I can come back to this later if anyone has a better suggestion, but I want to get this in since it's a huge change already. Everything is passing and I have Richard's approval. Still, I'm curious about @wamartin-aml 's opinion on this one.